akka-stream/src/main/scala/akka/stream/scaladsl/FileIO.scala

```diff
@@ -86,5 +104,21 @@ object FileIO {
    * @param options File open options, defaults to Set(WRITE, CREATE)
    */
   def toPath(f: Path, options: Set[StandardOpenOption] = Set(WRITE, CREATE)): Sink[ByteString, Future[IOResult]] =
-    new Sink(new FileSink(f, options, DefaultAttributes.fileSink, sinkShape("FileSink")))
+    toPath(f, options, startPosition = 0)
+
+  /**
+   * Creates a Sink which writes incoming [[ByteString]] elements to the given file path. Overwrites existing files by default.
+   *
+   * Materializes a [[Future]] of [[IOResult]] that will be completed with the size of the file (in bytes) at the streams completion,
+   * and a possible exception if IO operation was not completed successfully.
+   *
+   * This source is backed by an Actor which will use the dedicated `akka.stream.blocking-io-dispatcher`,
+   * unless configured otherwise by using [[ActorAttributes]].
+   *
+   * @param f the file path to write to
+   * @param options File open options, defaults to Set(WRITE, CREATE)
+   * @param startPosition the start position to write to
+   */
+  def toPath(f: Path, options: Set[StandardOpenOption], startPosition: Long): Sink[ByteString, Future[IOResult]] =
```

Comment:
I think it be changed to `options: Set[OpenOption]`. It would be source compatible and also binary compatible thanks to type erasure (should be verified)